weather_mv/loader_pipeline/sinks.py

```diff
@@ -354,35 +383,52 @@ def open_dataset(uri: str,
                  open_dataset_kwargs: t.Optional[t.Dict] = None,
                  disable_grib_schema_normalization: bool = False,
                  tif_metadata_for_datetime: t.Optional[str] = None,
+                 group_common_hypercubes: t.Optional[bool] = False,
                  band_names_dict: t.Optional[t.Dict] = None,
                  initialization_time_regex: t.Optional[str] = None,
-                 forecast_time_regex: t.Optional[str] = None) -> t.Iterator[xr.Dataset]:
+                 forecast_time_regex: t.Optional[str] = None) -> t.Iterator[t.List[xr.Dataset]]:
```

Comment:
Just thinking aloud: what are ways that we could implement this to get backwards compatibility? Is there a way we could introduce this change that would insulate us from breaking internal pipelines? 

One idea: What if we return a union type here (`t.Union[xr.Dataset, t.List[xr.Dataset]]`, just like the above)? For most dataset (e.g. at least all non-grib pipelines, and most grib pipelines) we'll use the first output of the union. This may save us from making code changes to a lot of areas of the codebase, for example, we won't have to implement this in our BQ code for now. 

It's not clear to me, for example, that the BQ pipeline here is correct. For this change, we'd need to test that this support creates the correct BQ schema as well as writes rows accurately. I'd say it's safer to assume all BQ paths that try to open lists of Datasets throw an error for now, and have this only apply to the EE pipeline (I'd probably be more comfortable with the EE pipeline AND when the `group_common_hypercubes` flag is set, which is more predictable). 

CC: @fredzyda -- LMK if this makes sense to you or if you have any thoughts here.

Reply:
I think you're right @alxmrs. I think maintaining backwards compatibility would be a good thing here.

Reply:
Have maintained backwards compatibility with the previous pipelines. The function will return a list of datasets only in case the user passes the `--group_common_hypercubes` flag.